lib/actions/logs.coffee

```diff
@@ -42,14 +30,15 @@ module.exports =
 	]
 	permission: 'user'
 	action: (params, options, done) ->
-		resin.logs.subscribe params.uuid, {
-			history: options.num or LOGS_HISTORY_COUNT
-			tail: options.tail
-		}, (error, message) ->
-			return done(error) if error?
-			if _.isArray(message)
-				_.each message, (line) ->
+		promise = resin.logs.history(params.uuid).each (line) ->
+			console.log(line.message)
+
+		if not options.tail
+			return promise.nodeify(done)
+
+		promise.then ->
+			resin.logs.subscribe(params.uuid).then (logs) ->
+				logs.on 'line', (line) ->
 					console.log(line.message)
-			else
-				console.log(message.message)
-			return done()
+				logs.on('error', done) ->
```

Comment:
There shouldn't be a `->` here, just:

```coffee
logs.on('error', done)
```

The point is that when an error happens, `done` is called, so we pass the function directly as an argument.

This is a shortened version of:

```coffee
logs.on 'error', (error) ->
    done(error)
```